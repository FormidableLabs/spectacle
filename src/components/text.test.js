import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Text from './text';

describe('<Text />', () => {
  it('should render a <p> with text for the default configuration.', () => {
    const context = { styles: { components: { text: { color: '#000' } } } };
    const wrapper = mount(<Text>Spectacle!</Text>, { context });
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should render a <div> with text for the `fit` configuration.', () => {
    const context = { styles: { components: { text: { color: '#000' } } } };
    const wrapper = mount(<Text fit>Spectacle Full Size!</Text>, { context });
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
