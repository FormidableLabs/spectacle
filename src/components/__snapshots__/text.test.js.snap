// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render a <div> with text for the \`fit\` configuration. 1`] = `
<Text
  fit={true}
  lineHeight={1}
>
  <Styled(div)
    className={undefined}
    height={0}
    innerRef={[Function]}
    styles={
      Object {
        "base": Object {},
        "context": Object {
          "color": "#000",
        },
      }
    }
  >
    <div
      className="css-1rapqq0"
      height={0}
    >
      <Styled(span)
        innerRef={[Function]}
        lineHeight={1}
        scale={NaN}
        styles={
          Object {
            "typeface": Object {},
            "user": undefined,
          }
        }
      >
        <span
          className="css-piqx9m"
        >
          Spectacle Full Size!
        </span>
      </Styled(span)>
    </div>
  </Styled(div)>
</Text>
`;

exports[`<Text /> should render a <p> with text for the default configuration. 1`] = `
<Text
  lineHeight={1}
>
  <Styled(p)
    className={undefined}
    styles={
      Object {
        "base": Object {},
        "context": Object {
          "color": "#000",
        },
        "typeface": Object {},
        "user": undefined,
      }
    }
  >
    <p
      className="css-cmzp7u"
    >
      Spectacle!
    </p>
  </Styled(p)>
</Text>
`;
